internal/security/secretstore/client/secretclient.go

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger"
+)
+
+const (
+	vaultToken = "X-Vault-Token" // nolint:gosec
+)
+
+var errNotFound = errors.New("secret NOT found")
+
+type SecretCollect struct {
+	Pair map[string]string `json:"data"`
+}
+
+type SecretClient struct {
+	client               internal.HttpCaller
+	rootToken            string
+	secretServiceBaseURL string
+	loggingClient        logger.LoggingClient
+}
+
+func NewSecretClient(
```

Comment:
are we able to reuse the secret client from go-mod-secret?

Reply:
Removed all tls package related files in this PR, will open another issue to support Postgres TLS connection.